### PR TITLE
Feature/pagination

### PR DESCRIPTION
**Pull request by capetillo**
### **Background:**
When the Dashboard first loaded, it would take a really long time to fetch for the thumbnails (and most of the time it would time out). 

### **Implementation:**
* `loadThumbnailsForPage`: This loads and displays thumbnails for sessions on the current page. It calculates the start and end indices based on `currentPage` and `pageSize`, clears the existing `thumbnailsMap`, and asynchronously fetches thumbnails for each session. This function will make the request to `getThumbnails` only 5 times per page (i.e. `pageSize`)
* `sessionsWithThumbnails`: Updated to filter and return sessions for the current page that have associated thumbnail urls. Aka only sessions with available thumbnails are displayed
* Added a watcher for `filteredSessions` because when the component first mounts it's an empty array and that would not trigger the cascade of `loadThumbnailsForPage` --> `getThumbnails`

### **Two other small fixes**
* The url to fetch pending requests was missing an `&` in the `obsPortalData` store
* When the user would log in, the log in button would stay until the user refreshed the site. That has been fixed by making the username a computed property in `App`

### VISUALS
**User logs in, username is on the right (rather than `log in` button) and thumbnails load quickly. This is also after clearing my cache so it's reliable!**

https://github.com/user-attachments/assets/5ef31a7c-dd9e-45c4-9d9d-e0fe5d7761f7

### CODE REVIEW

File: src/components/Images/MyGallery.vue [hunk 1]
```diff
@@ -24,35 +26,56 @@ const filteredSessions = computed(() => {
...
 })
 
+const totalPages = computed(() => {
+  return Math.ceil(filteredSessions.value.length / pageSize)
+})
+
 const getThumbnails = async (observationId) => {
   await fetchApiCall({
-    url: configurationStore.thumbnailArchiveUrl + `thumbnails/?observation_id=${observationId}&size=large`,
+    url: `${configurationStore.thumbnailArchiveUrl}thumbnails/?observation_id=${observationId}&size=large`,
```
Comment (jnation3406) [inline src/components/Images/MyGallery.vue:35]: Maybe I'm missing something, but if these thumbnails are for the page where it lists your sessions and all their little thumbnails under the dates then I think it should use the small thumbnails, not the large ones. The large ones are like 1/2 your screen size so I would save those for like opening a single image into a larger view...

**Review by jnation3406 — APPROVED** [review]
It kind of looks like anytime there is a change in session (new session added or whatever) it will try to reload the thumbnails for all sessions, not just the new ones added? I guess that's fine for now but we should make sure this doesn't make too many calls and take to long for people with large numbers of requests, or large numbers of thumbnails per request.

File: src/components/Images/MyGallery.vue (hunk 1 above)
Comment (capetillo) [reply src/components/Images/MyGallery.vue:35]: okey dokes! I should really start testing on the ipad

Comment (capetillo) [reply src/components/Images/MyGallery.vue]: and to respond to your top comment, I can definitely add the `limit` query param to the call. But I'm not sure what the solution could be. Maybe we talk about this in zoom

**Pull request merged**